File: 0-example/xor.py
key = bytearray('We could either watch it happen or be a part of it.', 'utf-8')


def encrypt(msg):
    ct = bytearray(msg, 'utf-8')
    for i in range(len(ct)):
        ct[i] ^= key[i % len(key)]
        pass
    return ct.decode('utf-8')


msg = "How many days until Christmas?"

print(msg)
ct = encrypt(msg)
print(ct)
print(encrypt(ct))

File: 0-example/xor.go
package main

import (
	"fmt"
)

const key = "We could either watch it happen or be a part of it."

func encrypt(msg string) (ct string) {
	for i := range msg {
		ct += string(msg[i] ^ key[i%len(key)])
	}
	return ct
}

func main() {
	msg := "How many days until Christmas?"
	fmt.Println(msg)
	ct := encrypt(msg)
	fmt.Println(ct)
	fmt.Println(encrypt(ct))
}
